content/en/docs/concepts/configuration/secret.md

```diff
@@ -339,9 +339,9 @@ files.
 
 When a secret being already consumed in a volume is updated, projected keys are eventually updated as well.
 Kubelet is checking whether the mounted secret is fresh on every periodic sync.
-However, it is using its local ttl-based cache for getting the current value of the secret.
+However, it is using its local cache propagated via watch for getting the current value of the secret.
```

Comment:
@wojtek-t as per the PR you referenced, IIUC, the detection strategy is configurable. So, do we need to explain something more here?

Reply:
/hold

@wojtek-t feel free to hold cancel if there is no more explanation needed here :)

Reply:
Yes, this is configurable. @wojtek-t, could you augment the doc to include that? Thanks!

Reply:
PTAL - hopefully it's better now.

Reply:
Overall LGTM. One more question (just for user convenience), is there any place documented how the detection strategy be configured ? If users want to change this strategy, they may not know where and how to configure this. @wojtek-t @yujuhong WDYT?

@mtaufen since kubelet flags are migrating to config files, are the fields of configure documented ? Thanks!

Reply:
Those are the constants:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/kubeletconfig/v1beta1/types.go#L47
This is the config field: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/kubeletconfig/v1beta1/types.go#L636

We didn't add flag for that.

I can add that information, but I don't really know how to do that in concise enough way. Hints?

Reply:
I think mentioning the config field is sufficient. We can point users to the kubelet configuration doc if there's one.

Reply:
>We can point users to the kubelet configuration doc if there's one.

Totally agreed. Let's wait for the response from @mtaufen . I think he has more information on the related document status.

Reply:
For the config file, we link directly to the types.go file from the docs page.
See: https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/.

Reply:
Thanks @mtaufen !  But I notice that link references a fixed release types.go 
(release 1.10: https://github.com/kubernetes/kubernetes/blob/release-1.10/pkg/kubelet/apis/kubeletconfig/v1beta1/types.go) while the current docs version is 1.11. Shouldn't we use the same versioned types.go with docs?

Reply:
It must not have been updated in the 1.11 docs.
I sent https://github.com/kubernetes/website/pull/10087 to update.

Reply:
@mtaufen thanks for the update! Review comment is in that PR :)

Reply:
I added the link to appropriate field to this doc.

PTAL